meter/sma.go

```diff
@@ -0,0 +1,92 @@
+package meter
+
+import (
+	"errors"
+	"time"
+
+	"github.com/andig/evcc/api"
+	"github.com/andig/evcc/meter/sma"
+)
+
+const (
+	udpTimeout = 10 * time.Second
+)
+
+// SMA supporting SMA Home Manager 2.0 and SMA Energy Meter 30
+type SMA struct {
+	uri        string
+	power      float64
+	lastUpdate time.Time
+	recv       chan sma.TelegramData
+}
+
+// NewSMAFromConfig creates a SMA Meter from generic config
+func NewSMAFromConfig(log *api.Logger, other map[string]interface{}) api.Meter {
+	sm := struct {
+		URI string
+	}{}
+	api.DecodeOther(log, other, &sm)
+
+	return NewSMA(sm.URI)
+}
+
+// NewSMA creates a SMA Meter
+func NewSMA(uri string) *SMA {
+	log := api.NewLogger("sma ")
+
+	sm := &SMA{
+		uri:  uri,
+		recv: make(chan sma.TelegramData),
+	}
+
+	if sma.Instance == nil {
+		sma.Instance = sma.New(log, sm.uri)
+	}
+
+	sma.Instance.Subscribe(uri, sm.recv)
+
+	sm.lastUpdate = time.Now()
+	go sm.receive()
+
+	return sm
+}
+
+// receive processes the channel message containing the multicast data
+func (sm *SMA) receive() {
+	for {
+		msg := <-sm.recv
+		if msg.Data == nil {
+			return
+		}
+
+		var powerIn, powerOut float64
+		foundValues := false
+		for _, element := range msg.Data {
+			if element.ObisCode == "1:1.4.0" {
```

Comment:
evtl. relevante Obis Codes als `const`?

Reply:
Wie wäre da das Naming am Besten?

```golang
const (
  ObisGridConsumePower = "1:1.4.0"
  ObisGridConsumeEnergy = "1:1.8.0"
  ObisGridFeedinPower = "1:2.4.0"
  ObisGridFeedinEnergy  = "1:2.8.0"
)
```

wäre descriptive, aber Go typisch?

Reply:
Denke Ja. Consume/Feed würde ich in Anlehnung am MBMD als Import/Export bezeichnen?

Reply:
Dann bräuchte es auch das „Grid“ nicht mehr da in Import/Export impliziert?